libgnucash/quotes/finance-quote-wrapper.in

```diff
@@ -0,0 +1,263 @@
+###                         Finanace::Quote. Only intended to be used
+###                         from gnucash code.
+### Based on code taken from gnc-fq-helper.
+### Copyright 2001 Rob Browning <rlb@cs.utexas.edu>
+### Copyright 2021 Geert Janssens
+### 
+### This program is free software; you can redistribute it and/or    
+### modify it under the terms of the GNU General Public License as   
+### published by the Free Software Foundation; either version 2 of   
+### the License, or (at your option) any later version.              
+###                                                                  
+### This program is distributed in the hope that it will be useful,  
+### but WITHOUT ANY WARRANTY; without even the implied warranty of   
+### MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    
+### GNU General Public License for more details.                     
+###                                                                  
+### You should have received a copy of the GNU General Public License
+### along with this program# if not, contact:
+###
+### Free Software Foundation           Voice:  +1-617-542-5942
+### 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+### Boston, MA  02110-1301,  USA       gnu@gnu.org
+######################################################################
+
+use strict;
+use English;
+
+=head1 NAME
+
+finance-quote-wrapper  -  internal interface between gnucash and Finance::Quote
+
+=head1 SYNOPSIS
+
+finance-quote-wrapper
+
+=head1 DESCRIPTION
+
+Input: a JSON encoded hash of namespaces and commodities to query prices for.
+Currencies all go under the "currency' namespace, other commodities are
+grouped according to the quotes source they should be queried from
+There should also be a "defaultcurrency" key with the currency to be used as
+base currency for currency quotes.
+
+{
+    "defaultcurrency": "EUR",
+    "currency": {
+        "XAG": "",
+        "HKD": "",
+        "USD": ""
+    },
+    "yahoo_json": {
+        "CSCO": ""
+    }
+}
+
+Output (on standard output):
+
+The retrieved quotes in JSON format for further processing. These are
+the raw values returned by Finance::Quote. The caller is responsible for
+parsing and interpreting the results.
+
+If there are program failures, an error message will be printed on standard error.
+
+Exit status
+
+0 - success
+non-zero - failure
+
+=cut
+
+sub check_modules {
+  my @modules = qw(Finance::Quote JSON::Parse Getopt::Std);
+  my @missing;
+
+  foreach my $mod (@modules) {
+    if (eval "require $mod") {
+      $mod->import();
+    }
+    else {
+      push (@missing, $mod);
+    }
+  }
+
+  return unless @missing;
+
+  print STDERR "\n";
+  print STDERR "You need to install the following Perl modules:\n";
+  foreach my $mod (@missing) {
+    print STDERR "  ".$mod."\n";
+  }
+
+  print STDERR "\n";
+  print STDERR "Use your system's package manager to install them,\n";
+  print STDERR "or run 'gnc-fq-update' as root.\n";
```

Comment:
Now that our quote handling code has been rewritten in C++, could this error message end up in an error dialog presented to the user ? I believe so, as our call to boost asio captures the output on stderr.
In that case we may have fine-tune the message as it's heavily linux oriented with references to "running commands as root "or using a "system package manager" to install perl modules.

Reply:
Indeed it can, meaning that we also need to translate it. Better then to put a short string on stderr that GncFQQuoteSource can interpret and emit the error message than to try to get gettext working on Perl.

As for corrective action, we can say to re-run the online quotes install tool on Windows, system package manager for Linux, and `sudo gnc-fq-update` for macOS.

Reply:
OK, this is done. I decided that dumping enough instruction to be useful for all OSes wouldn't make for a good error dialog or log message and opted for a wiki link instead.